src/client/shadows/dynamicshadowsrender.cpp

```diff
@@ -432,39 +435,37 @@ void ShadowRenderer::renderShadowMap(video::ITexture *target,
 	m_driver->setTransform(video::ETS_VIEW, light.getFutureViewMatrix());
 	m_driver->setTransform(video::ETS_PROJECTION, light.getFutureProjectionMatrix());
 
-	// Operate on the client map
-	for (const auto &shadow_node : m_shadow_node_array) {
-		if (strcmp(shadow_node.node->getName(), "ClientMap") != 0)
-			continue;
+	// No cached ID for clientmap, ignore
+	if (m_shadow_node_clientmap_cache_id == -1)
+		return;
 
-		ClientMap *map_node = static_cast<ClientMap *>(shadow_node.node);
+	auto shadow_node = m_shadow_node_array[m_shadow_node_clientmap_cache_id];
+	ClientMap *map_node = static_cast<ClientMap *>(shadow_node.node);
```

Comment:
this looks like a big hack anyway
why can't the ShadowRenderer directly retrieve the ClientMap?

Reply:
excellent question, i kept the code without going further, if we don't need shadow_node specially for client map we can directly call the clientmap and remove checks on both functions, i'll check

Reply:
ok i found a way to do this properly and it will drop this string comparison on runtime, thanks for the suggestion we now have proper code path for both map and other nodes